tsc/scripts/git_count.py

```diff
@@ -0,0 +1,495 @@
+#!/usr/bin/env python3
+
+import subprocess
+import os
+import pprint
+import json
+import time
+from datetime import datetime
+import argparse
+
+#
+# List of repros
+#
+repros = [
+    "SONiC",          "sonic-host-services",  "sonic-mgmt-framework",    "sonic-quagga",     "sonic-swss",         "sonic-ztp",
+    "sonic-dbsyncd",  "sonic-linux-kernel",   "sonic-platform-common",   "sonic-restapi",    "sonic-swss-common",  "sonic-buildimage",
+    "sonic-fips",     "sonic-mgmt",           "sonic-platform-daemons",  "sonic-sairedis",   "sonic-telemetry",
+    "sonic-frr",      "sonic-mgmt-common",    "sonic-py-swsssdk",        "sonic-snmpagent",  "sonic-utilities"
+]
+
+#repros = ["sonic-buildimage" ]
+
+#
+# Util function: debugprint
+#
+debug = False
+
+def debugprint(str):
+    global debug
+    if debug:
+        print(str
+        )
+#
+# Per company commit counts, companys for per repro counts, gcompanys for all repro's counts
+#
+companys={}
+gcompanys={}
+def update_company_commit_counts(comp, count =1):
+    global companys
+    global gcompanys
+
+    # Per company per repor
+
+    if comp in companys:
+        companys[comp] = companys[comp] +count
+    else:
+        companys[comp] = count
+
+    # Total per repro        
+    if "Total" in companys:
```

Comment:
The current logic resets companys array while looping through in an upper level API. This is very hard to read. I prefer indexing companys to have repo like companys[repo][comp] and then going over this dic to print at the end